astropy/io/fits/card.py

```diff
@@ -1272,29 +1272,21 @@ def _format_value(value):
     if isinstance(value, str):
         if value == "":
             return "''"
-        else:
-            exp_val_str = value.replace("'", "''")
-            val_str = f"'{exp_val_str:8}'"
-            return f"{val_str:20}"
+        exp_val_str = value.replace("'", "''")
+        val_str = f"'{exp_val_str:8}'"
+        return f"{val_str:20}"
 
+    val_str = None
     # must be before int checking since bool is also int
-    elif isinstance(value, (bool, np.bool_)):
```

Comment:
I think this is an example of a rule forcing worse code style - since every branch has an immediate return, what was there was super clear.  Not enough to nix this, but it adds to my feeling that we're starting to apply rules where common sense would be a better guide.

I think a change like you make below, where every branch that has a return has an `if` instead of an `elif` would be a better improvement.

Reply:
The one argument in favor of having a single return statement is that all the strings are formatted with `:>20` and the single return statement allows not having to repeat that.

Reply:
That's a fair argument! As I noted above, I don't feel strongly enough to nix the PR for it, but worried we were letting rules force styles. Given that you made the changes, I'm happy to go with them - would just suggest not to start enforcing the rule (and thus not have `noqa` for it).